### PR TITLE
Use node 18.18.2 and npm 10.2.3

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

Npm has caused a couple of lockfile-related issues recently. I think it makes sense to update npm just like we update the rest of our dependencies.

Related:

- https://github.com/pixiebrix/pixiebrix-extension/pull/6848
- https://github.com/pixiebrix/pixiebrix-extension/issues/6789
- https://github.com/pixiebrix/pixiebrix-extension/pull/6741

## Future Work

- [ ] #6741 

## Team Coordination

- [ ] This PR requires a node/npm version update: let the team know on #engineering

## Steps to take locally after merging

```sh
nvm use
npm install npm@10.2.3 --global
```

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version-file: ".nvmrc"
+          node-version-file: package.json
```
Comment (fregante) [inline .github/workflows/ci.yml:18]: The action now supports reading directly from package.json, we don't need to maintain another file

- https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file

File: .nvmrc [hunk 2]
```diff
@@ -1 +0,0 @@
-18.13
```
Comment (fregante) [inline .nvmrc:1]: I'm assuming this was just used by GitHub actions and is now redundant. I can restore it if necessary.

File: package.json [hunk 3]
```diff
@@ -23,8 +23,8 @@
   },
   "engine-strict": true,
   "engines": {
-    "node": ">=18.13 <19",
-    "npm": "8.x"
+    "node": ">=18.18.2 <19",
+    "npm": "10"
```
Comment (fregante) [inline package.json:27]: I also bumped Node to the latest **minor** version in v18 because npm 10 requires  `>=18.17`

File: package.json [hunk 4]
```diff
@@ -23,8 +23,8 @@
   },
   "engine-strict": true,
   "engines": {
-    "node": ">=18.13 <19",
-    "npm": "8.x"
+    "node": "18.18.2",
+    "npm": "10.2.3"
```
Comment (fregante) [inline package.json:27]: cc @grahamlangford a you suggested. Fixed versions

File: .nvmrc (hunk 2 above)
Comment (grahamlangford) [reply .nvmrc:1]: I believe most of us use nvm to manage our node versions, which requires the `.nvmrc`.

Comment (fregante) [reply .nvmrc:1]: Restore

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**